File: client/src/app/shared/models/marketGroupMst.ts
import { IMarketGroupDtl } from "./marketGroupDtl";

export interface IMarketGroupMst {
    id: number;
    marketCode: string;
    groupName: string;
    employeeId: number; 
    status: string; 
    marketGroupDtls:IMarketGroupDtl[];
}
 
export class MarketGroupMst implements IMarketGroupMst {
    id: number=0;
    marketCode: string;
    groupName: string;
    employeeId: number; 
    status: string=null; 
    marketGroupDtls:IMarketGroupDtl[];
}

File: Core/Entities/MarketGroupMst.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace Core.Entities
{
    public class MarketGroupMst:BaseEntity
    {
        public string GroupName { get; set; }
        public string Status { get; set; }
        public string MarketCode { get; set; }
        //public int EmployeeId { get; set; }
        //[ForeignKey("EmployeeId")]
        //public Employee Employee { get; set; }
        [ForeignKey("MstId")]
        public virtual IEnumerable<MarketGroupDtl> MarketGroupDtls { get; set; }
    }
}
